File: php/features.php
<?php
declare(strict_types=1);

// CurrencyExchange SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CurrencyExchangeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CurrencyExchangeBaseFeature();
            case "test":
                return new CurrencyExchangeTestFeature();
            default:
                return new CurrencyExchangeBaseFeature();
        }
    }
}

File: py/features.py
# CurrencyExchange SDK feature factory

from feature.base_feature import CurrencyExchangeBaseFeature
from feature.test_feature import CurrencyExchangeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CurrencyExchangeBaseFeature(),
        "test": lambda: CurrencyExchangeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
